client/packages/common/src/plugins/usePluginEvents.ts

```diff
@@ -0,0 +1,40 @@
+import { FnUtils } from '@common/utils';
+import { useState } from 'react';
+
+type Event<EventInput, EventResult> = (
+  input: EventInput
+) => Promise<EventResult>;
+
+export const usePluginEvents = <EventInput, EventResult>() => {
```

Comment:
Events are changed to be a prop of this hook, please see documentation, also upstream `isDirty` state is change to be a more generic 'state'